### DISCUSSION TITLE
Transferring stewardship of the project over to ShakaCode?

### DISCUSSION THREAD

**Discussion by justin808** [reactjs/react-rails#1200] [General]
@BookOfGreg, How do you feel about passing the stewardship of this project, including the Github repo and the Ruby Gem release, over to ShakaCode?

My team and I have been getting more involved in this project for the past few months. There's a lot of hard work ahead, including:

1. New releases that accommodate the right mix of support for older versions of other libraries.
2. Updates to the CI system and better tests
3. Updates for React 17 and 18, while still supporting older versions
4. Finishing cleanup of issues and PRs
5. Building updated example apps

If anybody questions if ShakaCode is qualified, look how we took over the stewardship of [Shakapacker](https://github.com/shakacode/shakapacker) from [rails/webpacker](https://github.com/rails/webpacker).

The goal of [ShakaCode](https://www.shakacode.com) is to be the best software agency for firms using Ruby on Rails with React, regardless of the libraries used. Thus, our interests are aligned with the community.

Note, we have no interest in moving over react-rails projects to react_on_rails unless it is in their best interests. ShakaCode has many clients that use Shakapacker without react_on_rails.

**Comment by hibachrach** [comment]
Can I ask what the motivating reason is for moving `react-rails` under the @ShakaCode namespace? Can all of the work proposed above be done under the existing @reactjs org?

**Comment by justin808** [comment]
I want to move it over because I'm investing company resources to maintain the library. I need admin privileges to grant access to ShakaCode.com team members to help me maintain the project.

If you look at the history of commits before I started contributing, not much was happening for quite some time. Take a look at this issue for some more history: https://github.com/reactjs/react-rails/issues/1129.

The interests of the community and ShakaCode are aligned. ShakaCode finds consulting clients, especially ones using React with Rails, from its open-source portfolio and thus has much at stake to do a great job in maintaining the project.

**Reply by hibachrach** [reply]
Okay--I'm generally fine with that as long as we’re making an above-board deal here: we (the community) hand administration of `react-rails` over to @ShakaCode in exchange for renewed investment in the tool. I just want to make it explicit that that’s the justification.

**Reply by BookOfGreg** [reply]
I can add as many members as necessary, and I have a responsibility to the org to add and remove contributors liberally. Maintainers are bound by the Facebook and ReactJS guidelines.

http://reactcommunity.org/
>Collaborators are added liberally by other owners and/or collaborators. They need only contribute some non-trivial commits and follow the project's contribution guidelines and Code of Conduct (if one exists).
and
https://github.com/facebook/react/blob/main/CODE_OF_CONDUCT.md#our-responsibilities
> Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.

If I see people contribute, Justin you now have the powers to write and merge your teams changes without my involvement. And I can add in people from your team who are contributing to be able to write by themselves also if asked or if I notice independently.
You have the power to display publicly in the repo here that you and others of your team are contributing and advertise your involvement from here and readme and docs and wiki and in issues/prs.
What additional power would you like?

**Reply by justin808** [reply]
Hi @BookOfGreg, all sounds good. 

> What additional power would you like?

1. I need to ship releases.
* https://rubygems.org/profiles/railsonmaui
* https://www.npmjs.com/~shakacode
2. I need to configure project settings, such as CI settings.

**Comment by BookOfGreg** [comment]
This project is a part of the wider ReactJS Community.
https://reactjs.org/community/support.html
http://reactcommunity.org/
There were stewards before me and there will be others after me. It's part of our duty to grow the ownership within ReactJS community.

I don't think changing ownership from ReactJS/Facebook to ShackaCode is something that is within anyone's power to give.
